src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -170,5 +177,95 @@ public RestClient FindRestClient(OperationGroup operationGroup)
             EnsureHeaderModels().TryGetValue(operation, out var model);
             return model;
         }
+
+        private void DecorateOperationGroup()
+        {
+            foreach (var operations in _codeModel.OperationGroups)
+            {
+                operations.ProviderName = _context.Configuration.OperationGroupMapping.ContainsKey(operations.Key) ? _context.Configuration.OperationGroupMapping[operations.Key] : ConstructOperationProviderName(operations);
+            }
+        }
+
+        private string ConstructOperationProviderName(OperationGroup operations)
+        {
+
+            string? providerName = "";
+            var request = GetBestMethod(operations);
+            bool adding = false;
+            if (request != null)
+            {
+                foreach (var segment in GetPathSegments(request.Path))
+                {
+                    var asSplit = segment.Split('/', StringSplitOptions.RemoveEmptyEntries);
+                    if (asSplit?.Length > 1 && asSplit.First().Equals("providers"))
+                    {
+                        adding = true;
+                        providerName = segment.Substring("providers".Length + 2).TrimEnd('/');
+                    }
+                    else if (adding)
+                    {
+                        providerName += segment.TrimEnd('/');
+                    }
+                }
+            }
+            return providerName.TrimEnd('/');
```

Comment:
shouldn't this be where we Log an error? since at this point we will not have the resourcetype for this operations group?

Reply:
Yes, but it looks like the CI attempts to test other RPs, like search for example, and this will throw an exception for all the cases that we have not custom defined yet. That is, not defined in the readme.md for that RP for those resources we can't parse (for example resources that only have gets)

Reply:
msg will be
           
            if (request == null || resourceName== "")
            {
                throw new ArgumentException("Could not set ResourceType for operations group " + operationsGroup.Key +
                "\nPlease try setting this value for this operations in the readme.md for this swagger in the operation-group-mapping section");
            }